src/Core/Settings.h

```diff
@@ -639,6 +639,7 @@ class IColumn;
     M(MsgPackUUIDRepresentation, output_format_msgpack_uuid_representation, FormatSettings::MsgPackUUIDRepresentation::EXT, "The way how to output UUID in MsgPack format.", 0) \
     M(UInt64, input_format_max_rows_to_read_for_schema_inference, 100, "The maximum rows of data to read for automatic schema inference", 0) \
     M(Bool, input_format_json_read_bools_as_numbers, true, "Allow to parse bools as numbers in JSON input formats", 0) \
+    M(Bool, format_protobuf_google_wrappers_special_treatment, true, "Allow usage of Google value wrappers for regular values", 0) \
```

Comment:
`input_format_protobuf_flatten_google_wrappers` and `output_format_protobuf_nullable_as_google_wrappers`.
What do you think?
Sorry for being so persistent with the naming, I would just like to be as specific as possible so the users are fully aware what's going on and so they have better control over parsing/serialising of the protobuf files.

Reply:
No worries, I appreciate the effort to make ClickHouse readable for the user ^^
To be completely explicit, even three settings could be introduced:
* `input_format_protobuf_flatten_google_wrappers`
* `input_format_protobuf_empty_google_wrappers_as_defaults`
* `output_format_protobuf_serialize_default_nullable_google_wrappers`
What do you think?

Reply:
`output_format_protobuf_nullable_as_google_wrappers` I think is enough with a nice description, just to have setting name of reasonable length 😄 

> `input_format_protobuf_empty_google_wrappers_as_nullable_defaults` 

for this to work, the google wrapper needs to be flattened because `str.value` will be assigned to `str` column if there is a value. I.e. second setting needs to include the first one, and that's why it made sense to me to name it simply `input_format_protobuf_flatten_google_wrappers` (again, with a good description of the behaviour it produces).
The only reason to have it as a separate setting is if someone would like to flatten wrappers but not want the specific behaviour of null/default differentiation which I don't think will happen - please correct me if I'm wrong. 
Also, because it's an opt-in setting, we can easily change it someone actually will want it.

Reply:
I will introduce two settings then, one for input and one for output, both false by default.
So somebody might actually want to use only this flattening setting to start interpreting Google wrappers as regular columns, not necessarily nullable. This could be the case if someone had some component in their system which had to output Google wrappers, for example to ensure compatibility with some component other than ClickHouse, but wanted to interpret these as regular values in ClickHouse. The nullable aspect would not be important here.
But you are right the nullable part of the input depends on this flattening part. I also don't see why anyone would want to switch it off while keeping the flattening part, there is no legacy behaviour that could be affected.
For the output, the way this change could affect someone would be if they really expected a null when a default could be present, for example in tests on the message content or length, or when a 2-byte overhead per default nullable row was significant for the network (unlikely, I guess).